x-pack/plugin/transform/qa/multi-node-tests/build.gradle

```diff
@@ -36,6 +36,11 @@ testClusters.javaRestTest {
   keystore 'bootstrap.password', 'x-pack-test-password'
   keystore 'xpack.security.transport.ssl.secure_key_passphrase', 'testnode'
   numberOfNodes = 3
+  // Node roles are configured this way in order to verify redirecting the transform request from the node lacking
+  // remote_cluster_client role to the node that is remote_cluster_client.
+  nodes."javaRestTest-0".setting 'node.roles', '[data,ingest,master]'
+  nodes."javaRestTest-1".setting 'node.roles', '[data,ingest,master,transform]'
+  nodes."javaRestTest-2".setting 'node.roles', '[data,ingest,master,transform,remote_cluster_client]'
```

Comment:
Adding the `transform` role here is fine for master, but might cause problems in 7.x.  The reason is that 7.x still uses the transport client for half of integration tests that are not REST tests, and the transform client only understands roles that are defined in core Elasticsearch.  I thought the fix would be simple, so tried it in #68226, and that fixed it for the external cluster tests (and end users too), but unfortunately there is a big problem with the _internal_ cluster tests - https://github.com/elastic/elasticsearch/issues/69693#issuecomment-788137123 has the details.

So you can leave the code like this in master but will need to plan what to do in the 7.x backport.  If _all_ the transform multi-node tests are REST tests then you might get away with what you've got here in 7.x.  But make sure you run the backport PR CI at least 3 times to confirm the client randomisation doesn't sometimes cause problems.  If some of the transform multi-node tests use the node/transport client then it will be necessary to do quite a lot more refactoring in the backport.